### PR TITLE
Xuejun/fix cvs 98110

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Add distinguish for arm plugin
 - Skip BATCH related operations for arm plugin

### Tickets:
 - 98110

### CODE REVIEW

File: src/bindings/c/tests/ie_c_api_test.cpp [hunk 1]
```diff
@@ -31,7 +31,11 @@ std::condition_variable condVar;
         std::string plugins_xml_std = TestDataHelpers::generate_ieclass_xml_path("plugins_win.xml");
     #endif
 #elif defined __APPLE__
+    #ifdef __aarch64__
+        std::string plugins_xml_std = TestDataHelpers::generate_ieclass_xml_path("plugins_apple_arm.xml");
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ie_c_api_test.cpp:35]: It need https://github.com/openvinotoolkit/testdata/pull/13 first merged.

Comment (ilyachur) [reply src/bindings/c/tests/ie_c_api_test.cpp:35]: Can we register the plugin in the runtime without strange plugins xml files?

File: src/bindings/c/tests/ie_c_api_test.cpp [hunk 2]
```diff
@@ -131,7 +135,11 @@ TEST(ie_core_register_plugin, registerPlugin) {
     IE_EXPECT_OK(ie_core_read_network(core, xml, bin, &network));
     EXPECT_NE(nullptr, network);
 
-    const char *plugin_name = "openvino_intel_cpu_plugin";
+    #ifdef __aarch64__
+        const char *plugin_name = "openvino_arm_cpu_plugin";
```
Comment (ilyachur) [inline src/bindings/c/tests/ie_c_api_test.cpp:139]: I think that plugin name can be a common definition for this file.

File: src/bindings/c/tests/ie_c_api_test.cpp (hunk 1 above)
Comment (zhaixuejun1993) [reply src/bindings/c/tests/ie_c_api_test.cpp:35]: @ilyachur The plugins xml files are the input parameter for api unite test "ie_core_create(plugins_xml, &core)", so I don't think we can register in runtime.

File: src/bindings/c/tests/ie_c_api_test.cpp (hunk 2 above)
Comment (zhaixuejun1993) [reply src/bindings/c/tests/ie_c_api_test.cpp:139]: how about this?
```
#define ARM_CPU_PLUGIN_NAME "openvino_arm_cpu_plugin"
#define INTEL_CPU_PLUGIN_NAME "openvino_intel_cpu_plugin"
```

Comment (ilyachur) [reply src/bindings/c/tests/ie_c_api_test.cpp:139]: Why not something like:
```suggestion
#ifdef __aarch64__
    #define CPU_PLUGIN_NAME "openvino_arm_cpu_plugin"
#else
    #define CPU_PLUGIN_NAME "openvino_intel_cpu_plugin"
#endif
```

Comment (zhaixuejun1993) [reply src/bindings/c/tests/ie_c_api_test.cpp:139]: Good idea, will do this change. Thanks!

File: src/bindings/c/tests/ie_c_api_test.cpp [hunk 3]
```diff
@@ -1311,6 +1323,9 @@ TEST(ie_infer_request_infer_async, inferAsyncWaitTime) {
 }
 
 TEST(ie_infer_request_set_batch, setBatch) {
+    #ifdef __aarch64__
+        GTEST_SKIP() << "Skipping 'setBatch' test for ARM";
+    #endif
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ie_c_api_test.cpp:1328]: arm plugin doesn't support setBatch().

File: src/bindings/c/tests/ov_compiled_model_test.cpp [hunk 4]
```diff
@@ -172,6 +172,9 @@ TEST_P(ov_compiled_model, get_property) {
 }
 
 TEST_P(ov_compiled_model, create_compiled_model_with_property) {
+    #ifdef __aarch64__
+        GTEST_SKIP() << "Skipping 'create_compiled_model_with_property' test for ARM";
+    #endif
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ov_compiled_model_test.cpp:177]: Some of below properties are not supported by arm plugin?

File: src/bindings/c/tests/ov_core_test.cpp [hunk 5]
```diff
@@ -205,6 +205,9 @@ TEST_P(ov_core, ov_core_set_property_invalid_number_property_arguments) {
 }
 
 TEST_P(ov_core, ov_core_set_property_enum_invalid) {
+    #ifdef __aarch64__
+        GTEST_SKIP() << "Skipping 'ov_core_set_property_enum_invalid' test for ARM";
+    #endif
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ov_core_test.cpp:210]: Skip it due to unsupported properties in arm plugin?

File: src/bindings/c/tests/ie_c_api_test.cpp (hunk 1 above)
Comment (riverlijunjie) [reply src/bindings/c/tests/ie_c_api_test.cpp]: If we don't modify plugins_apple.xml, let close testdata PR13?

**Comment by riverlijunjie** [comment]
@zhaixuejun1993 please resolve clang format issue.

File: src/bindings/c/tests/ie_c_api_test.cpp [hunk 6]
```diff
@@ -1369,6 +1388,9 @@ TEST(ie_infer_request_set_batch, setZeroBatch) {
 }
 
 TEST(ie_infer_request_set_batch, setNegativeBatch) {
+    #ifdef __aarch64__
```
Comment (ilya-lavrenov) [inline src/bindings/c/tests/ie_c_api_test.cpp:1391]: we must not streak the code with multiple ifdef else. Why can we make most of the tests to be generic?

Comment (ilya-lavrenov) [reply src/bindings/c/tests/ie_c_api_test.cpp:1391]: It means we should not skip tests like `ie_core_register_plugins`. Why are they not enabled for ARM? 
Tests must not rely on whether they are run on ARM or RISC-V platform, they must rely on plugin capabilities.

Comment (zhaixuejun1993) [reply src/bindings/c/tests/ie_c_api_test.cpp:1391]: @ilya-lavrenov will refactor this PR: 
1) Will create the test plugin.xml file in runtime, so the related APIs like 'ie_core_register_plugins' will enable for ARM plugin too.
2) Will move the related APIs like 'setNegativeBatch' (ARM plugin doesn't support it for now) together & use only one "#ifndef __aarch64__" to reduce multiple ifdef else using.
What do you think?

Comment (zhaixuejun1993) [reply src/bindings/c/tests/ie_c_api_test.cpp]: Had finished the refactor, pls help review. Test pass on windows/linux(intel cpu) & Mac OS(ARM cpu).

File: src/bindings/c/tests/ov_core_test.cpp [hunk 7]
```diff
@@ -203,7 +205,17 @@ TEST_P(ov_core, ov_core_set_property_invalid_number_property_arguments) {
...
+/*
+* Skipping the following tests for ARM CPU plugin:
+* 'ov_core_set_property_enum_invalid', cause the property doesn't support by ARM plugin
+* 'ov_core_set_and_get_property_enum', cause the property doesn't support by ARM plugin
+* 'ov_core_set_and_get_property_bool_invalid', cause the property doesn't support by ARM plugin
+* 'ov_core_set_get_property_str', cause the property doesn't support by ARM plugin
+* 'ov_core_set_multiple_common_properties', cause the property doesn't support by ARM plugin
+* 'ov_compiled_model_export_model', cause this feature doesn't support by ARM plugin
+* 'ov_core_import_model', cause this feature doesn't support by ARM plugin
+*/
```
Comment (ilya-lavrenov) [inline src/bindings/c/tests/ov_core_test.cpp:217]: - `ov_core_set_property_enum_invalid` - can we use `ov::hint::performance_mode` ? It's supported by ARM
- `ov_core_set_and_get_property_bool_invalid` - here we use enable profiling, it's supported by ARM.
maybe other tests with properties can also use common properties supported by ARM? The list of ARM supported properties is here https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/arm_plugin/src/arm_plugin.cpp#L151-L203

For Import / export tests like `ov_compiled_model_export_model` - can we check in test that optimization capabilities property does not contain import_export flag and skip tests in runtime? Here is how it's defined for ARM https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/arm_plugin/src/arm_plugin.cpp#L194-L200

File: src/bindings/c/tests/ov_compiled_model_test.cpp [hunk 8]
```diff
@@ -172,6 +172,10 @@ TEST_P(ov_compiled_model, get_property) {
 }
 
 TEST_P(ov_compiled_model, create_compiled_model_with_property) {
+#ifdef __aarch64__
+    GTEST_SKIP() 
+        << "Skipping 'create_compiled_model_with_property' test for ARM, cause the property doesn't support by ARM";
```
[the remark this answers is not in the document]
Comment (zhaixuejun1993) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:177]: will change the property to enable_profiling

File: src/bindings/c/tests/ov_core_test.cpp (hunk 7 above)
Comment (zhaixuejun1993) [reply src/bindings/c/tests/ov_core_test.cpp:217]: For 'ov_core_set_property_enum_invalid' & 'ov_core_set_and_get_property_bool_invalid', ARM plugin doesn't check the value, so for invalid value the interface also return true. (caused the test failed) I think can we add this checking in ARM plugin.
For 'ov_core_set_and_get_property_enum', ARM plugin only have one enum property ov::hint::performance_mode, but in the test case has two enum properties, can we remove one?
For 'ov_core_set_get_property_str', ARM plugin do not have string property, I think we need to skip this.
For 'ov_core_set_multiple_common_properties', which includes the string property caused the test failed.
For 'ov_compiled_model_export_model' & 'ov_core_import_model', will try your proposal.
How do u think @ilya-lavrenov

Comment (ilya-lavrenov) [reply src/bindings/c/tests/ov_core_test.cpp:217]: > For 'ov_core_set_property_enum_invalid' & 'ov_core_set_and_get_property_bool_invalid', ARM plugin doesn't check the value, so for invalid value the interface also return true. (caused the test failed) I think can we add this checking in ARM plugin.

Can we send a PR to ARM plugin to have correct checks?

> How do u think @ilya-lavrenov

Ok with the rest

Comment (zhaixuejun1993) [reply src/bindings/c/tests/ov_core_test.cpp:217]: get it! Will create an another PR for ARM plugin.

Comment (zhaixuejun1993) [reply src/bindings/c/tests/ov_core_test.cpp]: @ilya-lavrenov  According our alignments, had finished the change, pls help review.
**One thing need take care, this PR works need the PR (https://github.com/openvinotoolkit/openvino_contrib/pull/582) merged.**

File: src/bindings/c/tests/test_model_repo.hpp [hunk 9]
```diff
@@ -47,7 +48,46 @@ inline std::string generate_image_path(std::string dir, std::string filename) {
...
+#else
+    std::string libraryname = "libopenvino_intel_cpu_plugin.so";
+#endif
+    // Create the file
+    std::string plugin_xml = "plugin_test.xml";
+    std::ofstream plugin_xml_file(plugin_xml);
+
+    // Write to the file
+    plugin_xml_file << "<!--\n";
+    plugin_xml_file << "Copyright (C) 2020 Intel Corporation\n";
```
[the remark this answers is not in the document]
Comment (zhaixuejun1993) [reply src/bindings/c/tests/test_model_repo.hpp]: Fixed

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/c/tests/ov_core_test.cpp (hunk 7 above)
Comment (ilya-lavrenov) [reply src/bindings/c/tests/ov_core_test.cpp]: thanks!

**Pull request merged**